### ISSUE TITLE
Receive page

### ISSUE THREAD

**Issue by Realchacal** [monero-project/monero-gui#2381]
(Updated! Ignore changes from #2339, #2325, #2304, #2298.)

![image](https://user-images.githubusercontent.com/46682965/64520182-e1e1c180-d2f5-11e9-8a33-898f25765252.png)

Current GUI:
![image](https://user-images.githubusercontent.com/46682965/64475687-10d92580-d186-11e9-824c-710dafdc905b.png)

**Comment by tficharmers** [comment]
Interesting. You have made me think about this page. When you have multiple sub addresses, it isn't particularly obvious which sub address the QR code is associated with. So my thoughts are shown in the following edited screenshot.

1. to edit the name of a sub address, when you hover the mouse over the name, it changes to orange (perhaps a tooltip). When selected, the 'Set label' overlay appears.

**Comment by tficharmers** [comment]
Interesting. You have made me think about this page. When you have multiple sub addresses, it isn't particularly obvious which sub address the QR code is associated with. So my thoughts are shown in the following edited screenshot.

1. To edit the name of a sub address, when you hover the mouse over the name, it changes to orange (perhaps a tooltip appears at the bottom of the page). When selected, the 'Set label' text field overlay appears.
2. To copy an address, hover the mouse over the address and it changes to orange (perhaps a tooltip appears at the bottom of the page). When selected the address is copied to the clipboard. This is mirroring functionality on the 'Transactions' page.
3. There is a new QR code icon associated with each address that when selected, makes the QR code appear in an overlay (like the 'Set label' overlay).
4. I agree that the QR code buttons should be labelled with text. I would have them saying 'Copy' and 'Download'. Both in grey.

That's it.

![gui](https://user-images.githubusercontent.com/23356013/64474812-09a81c80-d172-11e9-96c3-a53a9fa3ee6e.png)

**Comment by Realchacal** [comment]
> 1. when you hover the mouse over the name, it changes to orange

Excellent idea! --> Proposal updated.

With regard to copying the address by clicking on it: It's suboptimal if you click on something just for fun and then it copies it to the clipboard deleting what you previously had in your clipboard. Also it's suboptimal that the user must find out how to copy the address to the clipboard and just can't see it - but it's the most important thing one want to do on this page, so a button is really not too much. Also it's suboptimal that clicking on the name edits the name but clicking on the address copies the address. - On the other hand, that solution would offer the lowest visual complexity, which is **very** nice!

**Comment by tficharmers** [comment]
I see your points. My points mirrored existing functionality from the Transactions page, but even there they aren't that obvious. Anyway, just ideas to put in the mix.

I think the bigger issue for me though, which is why I knocked up the concept, is the floating/disassociated QR code. What are your thoughts on making it obvious which address this QR code is related to? Or do you think a QR code icon/button to bring it up in an overlay?

**Comment by Realchacal** [comment]
> The copy address functionality mirrored existing functionality from the Transactions page

uuuuh - didn't think of that. I'd rather prefer your copy solution than not having consistency on all pages.

> What are your thoughts on making it obvious which address this QR code is related to?

Highlight it? Thousand ways to do so! :D But just highlighting the account number is not enough, 100% agreed.

Another problem I just encountered: My rename solution is bullshit. If an address is labeled with an empty string it doesn't work.

**Comment by tficharmers** [comment]
> Another problem I just encountered: My rename solution is probably bullshit. If an address is labeled with an empty string it doesn't work.

The '#1', '#2', etc, could also activate the 'Set label' overlay.

**Comment by Realchacal** [comment]
> The '#1', '#2', etc, could also activate the 'Set label' overlay.

thx for trying to help me out, but no, that's not intuitive.


--> CLOSED because my proposal sucks because it sucks if on some pages clicking on text copies it while on other pages clicking on text edits it.

**Comment by tficharmers** [comment]
Well, your ideas prompted my ideas. Perhaps both our ideas will prompt others?

**Comment by Realchacal** [comment]
Reopened because updated! (100% recommended now.)

**Comment by rating89us** [comment]
Copy button should have a copy icon ( ![image](https://user-images.githubusercontent.com/45968869/68658079-1449ae00-0535-11ea-829c-5b61631c285f.png) ) in it.

**Comment by Realchacal** [comment]
> Copy button should have a copy icon ( ![image](https://user-images.githubusercontent.com/45968869/68658079-1449ae00-0535-11ea-829c-5b61631c285f.png) ) in it.

Would you also say, Apple should use a copy icon additionally to the text?
![image](https://user-images.githubusercontent.com/46682965/68658559-f6c91400-0535-11ea-8e59-805d22aa992a.png)

**Comment by rating89us** [comment]
> Do you want the copy icon instead of the text or added to the text?

Added to the text, like this: Copy ![image](https://user-images.githubusercontent.com/45968869/68658982-c170f600-0536-11ea-9621-92a69842cede.png)
 It won't get much bigger.

**Comment by Realchacal** [comment]
so would you say Apple should add icons here additionally to the text?
![image](https://user-images.githubusercontent.com/46682965/68660774-1f530d00-053a-11ea-8e7b-3936fef58507.png)

**Comment by rating89us** [comment]
**Some ideas for the position of UI elements in the screen:**
- Since Monero addresses/subaddresses are never stored in the blockchain, they can be reused to receive many transactions from the same sender. Therefore displaying labels are important, so the receiver can remember "subaddress 1 is for Bob", "subaddress 2 is for Charlie", and so on...
- After receiving from many different senders, it is expected for a wallet to have accumulated dozens or hundreds of subaddresses. In a Bitcoin wallet, these addresses should never be reused and should be hidden from the user. But since this is Monero, these addresses should be stored and available in the wallet, therefore all previously used addresses should fit in this page.
- `Create new address` button and `QR code` have really important functions and should **always** be visible, even when there are hundreds of subaddresses being displayed in this page.

**Current design:**
- User with 17 subaddresses
- `QR code` + `Create new address` button hidden at the bottom

![8](https://user-images.githubusercontent.com/45968869/69585106-0db04180-0fdf-11ea-815e-5afd614caa87.gif)

**My suggestion:**
- `QR code` fixed on the right side
- Scrollable Addresses list 
- `Create new address` button fixed on the bottom (always visible)
- Add pagination

![image](https://user-images.githubusercontent.com/45968869/69585355-bf4f7280-0fdf-11ea-936c-12c044bed113.png)

My suggestion is inspired by Trezor wallet, which doesn't have labels, pagination nor a scrollable list of addresses (previously used addresses are collapsed).
![image](https://user-images.githubusercontent.com/45968869/69584215-f07a7380-0fdc-11ea-9d29-13497a7e14bf.png)

**Comment by Realchacal** [comment]
> * Scrollable and compact Addresses list with pagination

Scrollable is good, pages suck. Who likes to navigate through pages!? That's why nothing has pages anymore. (Google, Facebook...)

> * `Edit label` button near the label

Nice. That's already included in my design.

> * `QR code` fixed at the right side (always visible: user can see it updating when an address is selected)

Please no! This feature is barely used so let's not give it such a prominent place. Instead, let's emphasize the most used feature on this page (see my initial proposal).

> * `Title above QR code` indicates which address is included in it

Nice.

**Comment by Realchacal** [comment]
> Everyone that sends crypto from a mobile to a desktop wallet uses a QR code, even more if the crypto is Monero, which has lengthy addresses.

Exactly. Only everybody who sends from a mobile wallet to a desktop wallet uses QR code, while most of our users send to/from a desktop wallet to/from an exchange or other service.
